std/algorithm/mutation.d

```diff
@@ -1695,6 +1696,59 @@ enum SwapStrategy
     assert(arr == [10, 9, 8, 4, 5, 6, 7, 3, 2, 1]);
 }
 
+/**
+Checks whether a type is either an integral static array (e.g. int[2])
+or a tuple of two integral types
+*/
+private template isIntegralPair(T)
```

Comment:
That's a lot of work! Just do this inline:

```d
    static if (is(T == Tuple!(U1, U2), U1, U2))
        enum isIntegralPair = isIntegral!U1 && isIntegral!U2;
    else
        enum isIntegralPair = false;
}
```

We don't want to support static arrays.

No need even to define a separate template, just do this in the body:

`static assert(is(T == Tuple!(U1, U2), U1, U2) && isIntegral!(T[0]) && isIntegral!(T[1]),
    "Each offset must be an integral or a tuple of two integrals");`